src/smooth_priv.h

```diff
@@ -0,0 +1,131 @@
+/*
```

Comment:
please explain rationale for this file, i.e. why an extra file and not part of smooth.h

Reply:
These are definitions of "private" types that are not part of the "interface".  

To avoid the file, I could

(1) move them to smooth.h and add a comment.

(2) move these to the beginning of smooth.c, but I think that this is quite a long block of definitions.

To keep the "interface" clean, I suggest (2)

Reply:
Yes, (2) please.